### ISSUE TITLE
Could Tianshou provide some ICM benchmark results?

### ISSUE THREAD

**Issue by puyuan1996** [aai-institute/tianshou#537]
- [ ] I have marked all applicable categories:
    + [ ] exception-raising bug
    + [ ] RL algorithm bug
    + [ ] documentation request (i.e. "X is missing from the documentation.")
    + [ ] new feature request
- [x] I have visited the [source website](https://github.com/thu-ml/tianshou/)
- [x] I have searched through the [issue tracker](https://github.com/thu-ml/tianshou/issues) for duplicates
- [ ] I have mentioned version numbers, operating system and environment, where applicable:
  ```python
  import tianshou, torch, numpy, sys
  print(tianshou.__version__, torch.__version__, numpy.__version__, sys.version, sys.platform)
  ```
Hi, I see that Tianshou has a good implementation of [ICM](https://github.com/thu-ml/tianshou/blob/master/tianshou/policy/modelbased/icm.py), could you please provide some benchmark results when combining different RL base algorithms like A2C/PPO/DQN in Atari/VizDoom/MiniGrid environments?
Thank you very much.

**Comment by Trinkle23897** [comment]
@nuance1979

**Comment by nuance1979** [comment]
@puyuan1996 Sure. Please take a look at the README.md in https://github.com/thu-ml/tianshou/tree/master/examples/vizdoom. For D2_navigation task, PPO+ICM has a clear advantage over PPO only; for D3_battle task, PPO+ICM also beats PPO only but both of them are beaten by C51 by a large margin. 

I have tried it on Atari but didn't see much improvement. However, I didn't spend much time tuning hyper parameters, either. It's possible that a carefully tuned set of hyper parameters might show improvements. You can try it out by:
```
cd examples/atari
python3 atari_ppo.py --task "BreakoutNoFrameskip-v4" --icm-lr-scale 10
```
Or
```
python3 atari_dqn.py --task "BreakoutNoFrameskip-v4" --icm-lr-scale 10
```
Then play with the ICM parameters `--icm-lr-scale`, `--icm-reward-scale`, `--icm-forward-loss-weight`.

**Comment by puyuan1996** [comment]
Thank you very much for your reply.
1. Have you tested the performance of off-policy ppo + ICM on the [MiniGrid](https://github.com/maximecb/gym-minigrid) environment? If yes, can you provide some experimental results? I tried it myself and found that the performance of off-policy ppo + ICM is not significantly better than off-policy ppo, some seeds are better, some seeds are worse.
2. Regarding the relative weight of intrinsic reward and extrinsic reward: icm-reward-scale, what are your considerations for setting the default value to [0.01]( https://github.com/thu-ml/tianshou/blob/c248b4f87e46d8fca229f29d5cabb15211c842e9/examples/vizdoom/vizdoom_ppo.py#L77). Is there a mechanism to automatically adjust this hyper-parameter?

Best Wishes.

**Comment by nuance1979** [comment]
> Thank you very much for your reply.
> 
> 1. Have you tested the performance of off-policy ppo + ICM on the [MiniGrid](https://github.com/maximecb/gym-minigrid) environment? If yes, can you provide some experimental results? I tried it myself and found that the performance of off-policy ppo + ICM is not significantly better than off-policy ppo, some seeds are better, some seeds are worse.

No, I have not tested on MiniGrid myself. You findings sound plausible because there is no guarantee that ICM will always help.

> 2. Regarding the relative weight of intrinsic reward and extrinsic reward: icm-reward-scale, what are your considerations for setting the default value to [0.01](https://github.com/thu-ml/tianshou/blob/c248b4f87e46d8fca229f29d5cabb15211c842e9/examples/vizdoom/vizdoom_ppo.py#L77). Is there a mechanism to automatically adjust this hyper-parameter?

I set it to match the authors' code [here](https://github.com/pathak22/noreward-rl/blob/3e220c2177fc253916f12d980957fc40579d577a/src/constants.py#L16). You can search for its [appearances](https://github.com/pathak22/noreward-rl/search?q=PREDICTION_BETA). I don't know whether there is any principled way of adjusting this hyper-parameter. If you really want to get the BEST result on a specific task, I would suggest looking into hyper-parameter tuning tools (or as part of an AutoML toolset), such as [Optuna](https://optuna.org), [W&B Sweeps](https://wandb.ai/site/articles/hyperparameter-tuning-as-easy-as-1-2-3), [Microsoft NNI](https://nni.readthedocs.io/en/stable/hyperparameter_tune.html), just to name a few.

> 
> Best Wishes.

**Comment by puyuan1996** [comment]
Thanks a lot for you answers. Best Wishes.
